app/controllers/photos_controller.rb

```diff
@@ -1,6 +1,7 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
   before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
+  before_action :ensure_user_is_authorized, only: [:show]
```

Comment:
OK to add this before action; but where are you defining ensure_user_is_authorized? Is your app currently running and displaying photos? Refactor this to before_action { authorize @photo || Photo}

Reply:
@Cindellai This still needs refactoring. Choose consistent before actions